core/src/main/java/com/radixdlt/consensus/sync/BFTSync.java

```diff
@@ -240,8 +251,50 @@ public SyncResult syncToQC(HighQC highQC, @Nullable NodeId author) {
     return switch (vertexStore.insertQc(qc)) {
       case VertexStore.InsertQcResult.Inserted inserted -> {
         // QC was inserted, try TC too (as it can be higher), and then process a new highQC
-        highQC.highestTC().map(vertexStore::insertTimeoutCertificate);
-        this.pacemakerReducer.processQC(vertexStore.highQC());
+        final var tcAlsoInserted =
+            highQC.highestTC().map(vertexStore::insertTimeoutCertificate).orElse(false);
+
+        this.pacemakerReducer
+            .processQC(vertexStore.highQC())
+            .ifPresent(
+                roundUpdate -> {
+                  CertificateType certificateType;
+                  if (tcAlsoInserted) {
+                    // TC was inserted so its round must be higher than that of a QC, round change
+                    // cert type is then a TC
+                    certificateType = CertificateType.TC;
+                  } else {
+                    // QC was inserted, need to determine whether its corresponding vertex was a
+                    // timeout vertex or not
+                    final var isTimeoutVertex =
+                        vertexStore
+                            .getExecutedVertex(qc.getProposedHeader().getVertexId())
```

Comment:
it looks scary (almost as if we were doing a DB read in order to bump a metric) :P
AFAIU, this only ever goes to a mem store, so is probably not a performance concern. On a conceptual level though, isn't there any other way of obtaining this information from context?

Reply:
not really, I guess we could carry the timeout flag over to BFTHeader :thinking: but I think this is out of scope of this PR

Reply:
and yeah, this is just reading from memory